pkg/api/controller.go

```diff
@@ -1415,6 +1415,39 @@ func (c *Controller) AttachPolicyToUser(w http.ResponseWriter, r *http.Request,
 	writeResponse(w, r, http.StatusCreated, nil)
 }
 
+func (c *Controller) GetConfig(w http.ResponseWriter, r *http.Request) {
+	_, err := auth.GetUser(r.Context())
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, ErrAuthenticatingRequest)
+		return
+	}
+	var storageCfg apigen.StorageConfig
+	internalError := false
+	if !c.authorizeCallback(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.ReadConfigAction,
+			Resource: permissions.All,
+		},
+	}, func(_ http.ResponseWriter, _ *http.Request, code int, v interface{}) {
+		switch code {
+		case http.StatusInternalServerError:
+			writeError(w, r, code, v)
+			internalError = true
+		case http.StatusUnauthorized:
+			c.Logger.Debug("Unauthorized request to get storage config, returning partial config")
+		}
+	}) {
+		if internalError {
+			return
+		}
+	} else {
```

Comment:
Why is this check necessary? Is it that bad if we return a partial config even in case there was an internal error while checking the authorization?

Reply:
IMO it is.
As a user, seeing partial information due to authorization is one thing, but seeing partial information due to internal errors is inconsistent and unclear

Reply:
In the end, it depends on what this information is used for, and what the experience would be without it. For example, if it is used to display a hint about the storage namespace when creating a repo, I would prefer to still be able to create a repo even if it's not possible to display the hint, regardless of the reason.